### PR TITLE
macos: Add support for macos arm64

### PR DESCRIPTION
**Pull request by matte1**

### CODE REVIEW

**Review by ograff — APPROVED** [review]

File: setup_toolchains.py [hunk 1]
```diff
@@ -42,9 +49,9 @@
 
 def create_wrappers(toolchain_dir, root_dir):
   for toolchain in _AVAILABLE_TOOLCHAINS:
-    host_os_name = '' if toolchain['host_os'] == 'linux' else "-" + toolchain['host_os']
+    host_os_name = '' if toolchain['host_os'] == 'linux' else toolchain['host_os'] + "-"
     name = ('arm-gnu-toolchain-' +
-            f'{toolchain["version"]}-{toolchain["host_arch"]}{host_os_name}-{toolchain["target"]}')
+            f'{toolchain["version"]}-{host_os_name}{toolchain["host_arch"]}-{toolchain["target"]}')
```
Comment (agoessling) [inline setup_toolchains.py:54]: Was this a latent bug?

File: platforms/platforms.bzl [hunk 2]
```diff
@@ -3,12 +3,12 @@ load("@bazel_arm_toolchains//toolchains:toolchain_info.bzl", "ALL_CPU", "AVAILAB
...
     for toolchain in AVAILABLE_TOOLCHAINS:
         for cpu in ALL_CPU:
-            name = "{}-{}-{}".format(cpu, toolchain["target_os"], toolchain["version"])
+            name = "{}-{}-{}-{}".format(toolchain["host_os"], cpu, toolchain["target_os"], toolchain["version"])
             native.platform(
                 name = name,
                 constraint_values = [
                     "//platforms/cpu:{}".format(cpu),
-                    "@platforms//os:{}".format(toolchain["target_os"]),
+                    "@platforms//os:{}".format(toolchain["target_os"]).replace("darwin","macos"),
```
Comment (agoessling) [inline platforms/platforms.bzl]: Can you make `os_constraint_name` a temp variable above with a comment of why this replace is needed?

Comment (agoessling) [reply platforms/platforms.bzl]: And below.

**Review by agoessling — CHANGES_REQUESTED** [review]

File: setup_toolchains.py (hunk 1 above)
Comment (matte1) [reply setup_toolchains.py:54]: I think so but maybe you just intended to not include the host os name. cc @ograff

File: platforms/platforms.bzl (hunk 2 above)
Comment (matte1) [reply platforms/platforms.bzl]: done, let me know if you want to invert what @ograff and  I did here and use macos as the base instead of darwin

Comment (matte1) [reply platforms/platforms.bzl]: @ograff authored this by the way

**Review by agoessling — APPROVED** [review]

**Pull request merged**